src/Simple/JSON.purs

```diff
@@ -192,13 +190,10 @@ instance readFieldsCons ::
   , Row.Lacks name from'
   , Row.Cons name ty from' to
   ) => ReadForeignFields (Cons name ty tail) from to where
-  getFields _ obj = do
-    value :: ty <- withExcept' $ readImpl =<< readProp name obj
-    rest <- getFields tailP obj
-    let
-      first :: Builder (Record from') (Record to)
-      first = Builder.insert nameP value
-    pure $ first <<< rest
+  getFields _ obj =
```

Comment:
This just looks like operator soup to me. I'd probably break it out into more steps in a do instead of less. Could also consider trying `ado` here.

Reply:
oh, nice!